nixos/modules/hardware/opengl.nix

```diff
@@ -69,21 +69,50 @@ in
       package = mkOption {
         type = types.package;
         internal = true;
+        default = cfg.mesaPackage;
         description = lib.mdDoc ''
           The package that provides the OpenGL implementation.
+
+          The default is mesa's drivers which should cover all OpenGL-capable
+          hardware. If you want to use another mesa version, adjust
+          {option}`mesaPackage`.
         '';
       };
-
       package32 = mkOption {
         type = types.package;
         internal = true;
+        default = cfg.mesaPackage32;
         description = lib.mdDoc ''
-          The package that provides the 32-bit OpenGL implementation on
-          64-bit systems. Used when {option}`driSupport32Bit` is
-          set.
+          Same as {option}`package` but for the 32-bit OpenGL implementation on
+          64-bit systems. Used when {option}`driSupport32Bit` is set.
         '';
       };
 
+      mesaPackage = mkOption {
+        type = types.package;
+        default = pkgs.mesa_23;
+        defaultText = literalExpression "pkgs.mesa_23";
+        example = literalExpression "pkgs.mesa_22";
+        description = lib.mdDoc ''
+          The mesa driver package used for rendering support on the system.
+
+          You should only need to adjust this if you require a newer mesa
```

Comment:
Nit: not really the case? You can totally want to update for newer features and such.

Reply:
Reasoning is that if you require newer features, you require a new mesa in my view.